### PR TITLE
Added matplotlib and csv export for data, as well as a way to track percentage of optimal payoffs selected

### PR DESCRIPTION
**Pull request by dalmeidad**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.ipynb_checkpoints
+*.ipynb
+results.csv
```
Comment (knospew2) [inline .gitignore]: we may also want to include `__pycache__`

File: main.py [hunk 2]
```diff
@@ -1,17 +1,61 @@
...
 
 def main():
 
-    # Command Line Arguments
-    script_name = sys.argv[0]
-    dataset_location = sys.argv[1]
-    algorithm_name = sys.argv[2]
-    time_steps = sys.argv[3]
+    # read commandline arguments, first
+    full_cmd_arguments = sys.argv
```
Comment (knospew2) [inline main.py]: This is great! We should be able to get a ton of flexibility here too in terms of the other hyperparameters for the ucms, agents, etc.

File: main.py [hunk 3]
```diff
@@ -20,14 +64,33 @@ def main():
     # The list of results
     results = []
 
+    # Percentage of optimal payoffs
```
Comment (knospew2) [inline main.py]: I think this is the count of optimal payoffs here and the percentage is later?

File: main.py [hunk 4]
```diff
@@ -1,17 +1,61 @@
...
-    algorithm_name = sys.argv[2]
-    time_steps = sys.argv[3]
+    # read commandline arguments, first
+    full_cmd_arguments = sys.argv
+    args = commandLine(full_cmd_arguments)
+    script_name = args['s']
+    dataset_location = args['d']
+    algorithm_name = args['a']
+    time_steps = args['t']
+    filename = args['f']
```
Comment (knospew2) [inline main.py]: Is filename the output filename? Could we rename this to be more descriptive?

Comment (dalmeidad) [reply main.py]: yes

**Pull request merged**